File: frontend/types/purchases.ts
export type PurchaseListItem = {
  purchase_id: string;
  purchase_no: string;
  purchase_date: string;
  supplier_id: string;
  supplier_name: string;
  reference_no: string;
  subtotal: number;
  status: string;
  created_at: string;
};

export type PurchaseLine = {
  line_id: string;
  variant_id: string;
  product_id: string;
  product_name: string;
  qty: number;
  unit_cost: number;
  line_total: number;
};

export type PurchaseDetail = PurchaseListItem & {
  note: string;
  created_by_user_id: string;
  lines: PurchaseLine[];
};

export type PurchaseLookupProduct = {
  variant_id: string;
  product_id: string;
  label: string;
  current_stock: number;
  default_purchase_price: number;
  sku: string;
  barcode: string;
};

export type PurchaseLookupSupplier = {
  supplier_id: string;
  name: string;
};

export type PurchaseCreatePayload = {
  purchase_date: string;
  supplier_id: string;
  reference_no: string;
  note: string;
  payment_status: 'paid' | 'unpaid' | 'partial';
  lines: Array<{ variant_id: string; qty: number; unit_cost: number }>;
};

File: easy_ecom/api/schemas/purchases.py
from pydantic import BaseModel, Field


class PurchaseLineRequest(BaseModel):
    variant_id: str = Field(min_length=1)
    qty: float = Field(gt=0)
    unit_cost: float = Field(gt=0)


class PurchaseCreateRequest(BaseModel):
    purchase_date: str = Field(min_length=1)
    supplier_id: str = ""
    reference_no: str = ""
    note: str = ""
    lines: list[PurchaseLineRequest] = Field(min_length=1)
    payment_status: str = Field(default="unpaid")


class PurchaseLineResponse(BaseModel):
    line_id: str
    product_id: str
    variant_id: str = ""
    product_name: str
    qty: float
    unit_cost: float
    line_total: float


class PurchaseSummary(BaseModel):
    purchase_id: str
    purchase_no: str
    purchase_date: str
    supplier_id: str = ""
    supplier_name: str = ""
    reference_no: str = ""
    subtotal: float
    status: str
    created_at: str


class PurchasesListResponse(BaseModel):
    items: list[PurchaseSummary]


class PurchaseDetailResponse(PurchaseSummary):
    note: str = ""
    created_by_user_id: str = ""
    lines: list[PurchaseLineResponse]


class PurchaseCreateResponse(BaseModel):
    purchase_id: str
    purchase_no: str
    subtotal: float
    status: str


class PurchaseLookupProduct(BaseModel):
    variant_id: str
    product_id: str
    label: str
    current_stock: float
    default_purchase_price: float = 0
    sku: str = ""
    barcode: str = ""


class PurchaseLookupSupplier(BaseModel):
    supplier_id: str
    name: str


class PurchaseFormOptionsResponse(BaseModel):
    products: list[PurchaseLookupProduct]
    suppliers: list[PurchaseLookupSupplier]
